sdk/src/main/java/com/google/cloud/dataflow/sdk/options/GcpOptions.java

```diff
@@ -183,24 +184,67 @@ void setCredentialFactoryClass(
   public static class DefaultProjectFactory implements DefaultValueFactory<String> {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultProjectFactory.class);
 
+    private static String getUserHome() {
+      return System.getProperty("user.home");
+    }
+
+    private String getActiveConfig() throws IOException {
+      final String defaultConfigName = "default";
+
+      // get active configuration
+      File activeConfFile = new File(
+          getUserHome(),
+          ".config/gcloud/active_config");
```

Comment:
I would wait for google-cloud-java, it seems as though they are working towards the change as evidenced on GoogleCloudPlatform/google-cloud-java#1207